Android UI: lots of cleanups, mainly around how Fragments are handled

hackbar opened the pull request:
This has a lot of stuff in it, but it's mostly around better abstraction with the emulation lifecycle, and using more standard Android features (eg animators).

lioncash commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/fragments/MenuFragment.java:
```diff
@@ -42,7 +53,11 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
-		mTitleText = (TextView) rootView.findViewById(R.id.text_game_title);
+		TextView titleText = rootView.findViewById(R.id.text_game_title);
+		String title = getArguments().getString(KEY_TITLE);
+		if (title != null){
```
brace should go on the next line.

lioncash commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/nVidiaShieldWorkaroundView.java:
```diff
@@ -0,0 +1,26 @@
+ */
+public final class nVidiaShieldWorkaroundView extends View
+{
+
```
unnecessary newline

lioncash commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/nVidiaShieldWorkaroundView.java:
```diff
@@ -0,0 +1,26 @@
+/**
+ * Work around a bug with the nVidia Shield.
+ */
+public final class nVidiaShieldWorkaroundView extends View
```
Classes should generally start with an upper-case letter.

lioncash commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java:
```diff
@@ -463,12 +398,13 @@ private void showScreenshot()
 		}
 	};
 
+
```
Unnecessary newlines?

hackbar commented:
Thanks for the review; I've fixed all these.

hackbar commented:
Hold off on merging this, I just found a bug with rotation and the controls overlay. I don't know if it's new, but it should be fixed.

hackbar commented:
I've rebased and added one more commit to fix a minor bug where the input overlay would race between Wii or Gamecube. There's still a bug with the overlays disappearing on rotation, but considering rotation is completely broken in HEAD, this change is still a positive.

I'm going to keep working at that, but I think it might be a framework bug. For the meantime, can you take another look at this?

hackbar commented:
@degasus can you merge this?

hackbar commented on Source/Android/app/build.gradle:
```diff
@@ -1,8 +1,8 @@
-    compileSdkVersion 25
-    buildToolsVersion '25.0.2'
+    compileSdkVersion 26
+    buildToolsVersion '26.0.1'
```
Probably update after this PR was created. Updated.

hackbar commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/TvMainActivity.java:
```diff
@@ -33,11 +34,11 @@
 import org.dolphinemu.dolphinemu.viewholders.TvGameViewHolder;
 
-public final class TvMainActivity extends Activity implements MainView
+public final class TvMainActivity extends FragmentActivity implements MainView
```
No, that would just make things more complicated, not simpler. Just pass around the base Activity (be is AppCompat* or Fragment*, depending on if you have to interact with leanback [which has no AppCompat]).

Etualbadusername commented:
How does one use the android version?
